File: ts-playwright-bdd/steps/pages/SearchPage.ts
import { Page, expect } from '@playwright/test';

export class SearchPage {
    readonly page: Page;
    readonly searchInput = "input[name='q']";
    readonly baseUrl = 'https://sauce-demo.myshopify.com/';

    constructor(page: Page) {
        this.page = page;
    }

    async goto() {
        await this.page.goto(this.baseUrl);
    }

    async searchFor(term: string) {
        await this.page.goto(this.baseUrl + '/search');
        await this.page.fill(this.searchInput, term);
        await this.page.press(this.searchInput, 'Enter');
    }

    async verifyTitle(text: string) {
        await expect(this.page).toHaveTitle(new RegExp(text));
    }
}

File: python-playwright-bdd/pages/search_page.py
import re
from playwright.sync_api import Page, expect
from utils.base_page import BasePage

class SearchPage(BasePage):
    BASE_URL = "https://sauce-demo.myshopify.com/"
    SEARCH_INPUT = "input[name='q']"

    def __init__(self, page: Page):
        super().__init__(page)

    def goto(self):
        self.navigate(self.BASE_URL)

    def search_for(self, term: str):
        self.navigate(f"{self.BASE_URL}/search")
        self.fill(self.SEARCH_INPUT, term)
        self.press(self.SEARCH_INPUT, "Enter")

    def verify_title(self, text: str):
        expect(self.page).to_have_title(re.compile(text))
